### ISSUE TITLE
Restore the Haskell Wiki theme

### ISSUE THREAD

**Issue by chreekat** [haskell/haskell-wiki-configuration#56]
With the crash upgrade that recently happened, the wiki is running the default Vector/2022 theme. We should tweak it to restore the old theme, or otherwise make it visually distinctive!

NB: I have not confirmed it, but I think tweaks to the vector theme can be made as regular edits at
https://wiki.haskell.org/MediaWiki:Vector.css

**Comment by simonmichael** [comment]
Here's that LLM answer, FWIW:

> That line of output:
> 
> ```html
> <templatestyles src="Template:Main_page/styles.css" />
> ```
> 
> appears because the MediaWiki software is not processing the `<templatestyles>` tag as intended. This tag is part of the **TemplateStyles** extension, which allows CSS styles to be defined in specific wiki pages (like `Template:Main page/styles.css`) and then included in other pages. When the raw tag is displayed, it means the TemplateStyles extension is likely not correctly enabled or configured on your MediaWiki installation.
> 
> Here's a breakdown of what could cause this and how to fix it:
> 
> **Cause:**
> 
> The most common cause is that the **TemplateStyles extension is not enabled** in your MediaWiki's configuration. The MediaWiki parser encounters the tag, doesn't recognize it as a valid instruction to process, and therefore just outputs the tag literally as part of the page content.
> 
> Less common causes could include:
> 
> * Errors in the TemplateStyles extension files themselves (less likely if installed correctly).
> * Severe parsing errors elsewhere on the page that prevent subsequent tags from being processed (unlikely for just this single tag).
> * Caching issues where an old version of the page or parser output is being served.
> 
> **How to Fix It:**
> 
> You have two main approaches:
> 
> 1.  **Enable the TemplateStyles Extension (Recommended):** This is the proper fix, as it allows the feature to work as intended and apply the styles defined in `Template:Main page/styles.css`.
> 
>     * **Access `LocalSettings.php`:** You need administrative access to your web server where MediaWiki is installed to edit the `LocalSettings.php` file. This file is usually located in the root directory of your MediaWiki installation.
>     * **Edit `LocalSettings.php`:** Open `LocalSettings.php` in a text editor. Find the section where extensions are typically loaded (often near the end of the file).
>     * **Add the loading line:** Add the following line:
>         ```php
>         wfLoadExtension( 'TemplateStyles' );
>         ```
>         Make sure this line is not commented out (doesn't start with `//` or `#`).
>     * **Save the file:** Save the changes to `LocalSettings.php`.
>     * **Run `update.php` (if necessary):** Depending on your MediaWiki version and how you manage updates, you might need to run the `maintenance/update.php` script from your server's command line to update the database schema for the new extension.
>     * **Clear MediaWiki Cache:** Even after enabling the extension, the old parsed version of the page might be cached. You can clear the cache by:
>         * Visiting the affected page and adding `?action=purge` to the URL (e.g., `https://wiki.haskell.org/Main_Page?action=purge`).
>         * Running the `maintenance/run.php` script with the `purge` job from the command line (requires shell access).
> 
>     After enabling the extension and clearing the cache, the `<templatestyles>` tag should be processed by MediaWiki, and the styles from `Template:Main page/styles.css` should be applied to the page instead of the raw tag being displayed.
> 
> 2.  **Hide the Output Using CSS (Workaround if Enabling is Not Possible):** If you cannot enable the TemplateStyles extension for some reason (e.g., lack of server access, wiki policy), you can hide the raw output using CSS. This will make the line disappear, but the intended styles will *not* be applied.
> 
>     * **Access Site-Wide CSS:** You need an account with interface administrator privileges on the wiki to edit site-wide CSS pages.
>     * **Edit `MediaWiki:Common.css`:** Go to the page `MediaWiki:Common.css` on your wiki (e.g., `https://wiki.haskell.org/MediaWiki:Common.css`). This page applies CSS to all users and all skins.
>     * **Add CSS to hide the tag:** Add the following CSS rule:
>         ```css
>         templatestyles {
>           display: none;
>         }
>         ```
>         This rule targets any HTML element with the tag name `templatestyles` and sets its display property to `none`, making it invisible.
>     * **Save the page:** Save the `MediaWiki:Common.css` page.
>     * **Bypass Browser Cache:** You might need to bypass your browser's cache to see the change immediately (usually Ctrl+Shift+R or Cmd+Shift+R).
> 
> This CSS workaround will hide the visible tag output, but the page will render without the styles intended by the template.
> 
> **In summary:** The most likely fix is to enable the TemplateStyles extension by adding `wfLoadExtension( 'TemplateStyles' );` to `LocalSettings.php` and clearing the cache. If that's not possible, you can hide the output by adding `templatestyles { display: none; }` to `MediaWiki:Common.css`. The first method is preferred for proper functionality.

**Comment by dynamDude** [comment]
> sm: currently it gives the impression Haskellers don't care much about their wiki
> sm: (has been broken since some time in 2024)
>
> [@davean](https://github.com/davean): Thats an hgolden thing, he's the wiki person
> 
> sm: To be clear, the "easy fix" I'm suggesting is to add "templatestyles { display: none; }" at https://wiki.haskell.org/MediaWiki:Common.css . I've updated the issue, hopefully hgolden will see this

chreekat has recently been organising myself and an interested few others. Currently the effort is going towards getting a vm specified using the current Nix setup to test changes - so please know the Wiki will be receiving love but the first steps won't result in immediately visible changes.

**Comment by chreekat** [comment]
Fwiw, I tried adding the `display: none` option in the css file, and it had no effect.

**Comment by simonmichael** [comment]
I see why: it's not actually an element, but literal text:
```html
<p><br />
&lt;templatestyles src="Template:Main_page/styles.css" /&gt;
</p>
```

Thanks for trying it.

**Comment by chreekat** [comment]
Yeah this is harder than it looks. There's a three-way conflict between the MediaWiki software version, the underlying config (found in this repo), and the UI-level config. All because the wiki was already hurting badly for available administration and everything was on an ancient version of everything. The failed system upgrade just blew everything up, and now there's nobody who knows how to fix it. It's on my list---hell, it's on everybody's list. But so are a lot of other things. The main problem is that there's no clear public visibility into (1) list of problems to fix (2) resourcing available to fix it. We, and especially @jmct , are aware of this problem and taking baby steps towards fixing it!

**Comment by simonmichael** [comment]
Wonderful, thanks all!
